src/fastfilter.h

```diff
@@ -23,24 +23,25 @@ constexpr bool isPow2(unsigned int num) { return num && !(num & (num - 1)); }
  *
  * This class can be used anywhere a Bloom filter is used so long as the input data is random.
  *
- * If NUM_HASH_FNS is 16 and FILTER_SIZE is >= 64k all bits in the uint256 input data will be used to set bits in
- * the filter.  If these fields are set to lower numbers, fewer bits may be used (although in the NUM_HASH_FNS case
- * execution will be faster).  Therefore, if this structure is used in an application that accepts externally created
- * uint256 numbers that are sensitive to deliberately constructed collisions, be sure to keep NUM_HASH_FNS high enough
- * that the creation of collisions in the used bits is not feasible.
- *
- * Note also that the input bits are used without obfuscation or mixing so if an attacker can control some input bits
- * the attacker can cause collisions in some of the fast filter entries for his inputs.  This will cause higher false
- * positive rates for these transactions.  For example, if the attacker can control 32 bits of the input, he can
- * effectively reduce the number of hash functions in the fast filter by 2 because he has engineered a guaranteed
- * collision for the two functions that use those bits.
- *
+ * For a FILTER_SIZE <=64k, the number of bits of the 256-bit input the filter uses is (NUM_HASH_FNS *
+ * log2(FILTER_SIZE)). Note that this means that in this regime and for low values for NUM_HASH_FNS, it becomes
+ * comparatively easier for an attacker to create collisions on purpose. It is expected that an attacker that has enough
+ * processing power to control <n> bits of the input key can reduce the effective number of "hash function equivalents"
+ * by about <n> / log2(FILTER_SIZE). Therefore, if this structure is used in an application that accepts externally
+ * created uint256 numbers that are sensitive to deliberately constructed collisions, be sure to keep NUM_HASH_FNS high
+ * enough that the creation of a large number of collisions in the used bits is not feasible.
+
+ * Conversely, for a FILTER_SIZE > 64k a number of ((NUM_HASH_FNS/2) * (log2(FILTER_SIZE)-16) ) bits are reused for
+ * accessing the filter's table. This will correspondingly reduce the filter's security. Furthermore, properties of the
```

Comment:
This reduction claim is not true (if I had an infinite FILTER_SIZE it would not be trivial to create a collision -- you'd still need to collide 256 bits), and the use of "security" is problematic since collision resistance != security.   How about this: "the filter's collision resistance is based on a maximum of 256 input bits and so a FILTER_SIZE > 64k cannot *increase* the collision resistance beyond this value."